engine/gameobject/src/gameobject/test/factory/test_gameobject_factory.cpp

```diff
@@ -160,7 +160,7 @@ TEST_F(FactoryTest, Factory)
     for (int i = 0; i < count; ++i)
     {
         uint32_t index = dmGameObject::AcquireInstanceIndex(m_Collection);
```

Comment:
Is this line needed if the index is no longer used?

Reply:
In this particular case it is not necessary, but in general we use index not just for name, but also to check if we can spawn an instance like
```
(index != dmGameObject::INVALID_INSTANCE_POOL_INDEX)
```
and then we assign index to the instance:
```
dmGameObject::AssignInstanceIndex(index, spawned_instance);
```

I can remove it in cases we don't call `AssignInstanceIndex`

Reply:
Ah, got it. The AcquireInstanceIndex() name is a bit confusing then if we no longer care about the index it returns? Instead of Pushing and Popping indices it seems like we should have an uint32_t counter or something? Or am I misunderstanding something?

Reply:
You can return them in random order. So it can't be a counter.

Reply:
Keep them!